control-plane-operator/hostedclusterconfigoperator/controllers/machine/machine.go

```diff
@@ -44,34 +36,45 @@ func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	switch hcp.Spec.Platform.Type {
 	case hyperv1.KubevirtPlatform:
 		if hcp.Spec.Platform.Kubevirt != nil && (hcp.Spec.Platform.Kubevirt.BaseDomainPassthrough != nil && *hcp.Spec.Platform.Kubevirt.BaseDomainPassthrough) {
-			if err := r.reconcileKubevirtDefaultIngressEndpoints(ctx, hcp, machine); err != nil {
-				return reconcile.Result{}, fmt.Errorf("failed reconciling kubevirt default ingress passthrough service endpoints: %w", err)
+			kubevirtPassthrowServices, err := r.findKubevirtPassthrowServices(ctx, hcp)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed finding kubevirt passthrow services")
+			}
+			for _, kubevirtPassthrowService := range kubevirtPassthrowServices {
+				if err := r.reconcileKubevirtPassthrowServiceEndpoints(ctx, hcp, req.NamespacedName, &kubevirtPassthrowService); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed reconciling kubevirt infra passthrow service endpoint slices")
+				}
```

Comment:
in order to not conflict with the kccm endpoint slice garbage collection I proposed, we shouldn't process a service with DeletionTimestamp != nil.